collect_test.go

```diff
@@ -0,0 +1,79 @@
+package goshopify
+
+import (
+	"reflect"
+	"testing"
+
+	httpmock "gopkg.in/jarcoal/httpmock.v1"
+)
+
+func collectTests(t *testing.T, collect Collect) {
+
+	// Test a few fields
+	cases := []struct {
+		field    string
+		expected interface{}
+		actual   interface{}
+	}{
+		{"ID", 18091352323, collect.ID},
+		{"CollectionID", 241600835, collect.CollectionID},
+		{"ProductID", 6654094787, collect.ProductID},
+		{"Featured", false, collect.Featured},
+		{"SortValue", "0000000001", collect.SortValue},
+	}
+
+	for _, c := range cases {
+		if c.expected != c.actual {
+			t.Errorf("Collect.%v returned %v, expected %v", c.field, c.actual, c.expected)
+		}
+	}
+}
+
+func TestCollectList(t *testing.T) {
+	setup()
+	defer teardown()
+
+	httpmock.RegisterResponder("GET", "https://fooshop.myshopify.com/admin/collects.json",
+		httpmock.NewStringResponder(200, `{"collects": [{"id":1},{"id":2}]}`))
+
+	collects, err := client.Collect.List(nil)
+	if err != nil {
+		t.Errorf("Collect.List returned error: %v", err)
+	}
+
+	expected := []Collect{{ID: 1}, {ID: 2}}
+	if !reflect.DeepEqual(collects, expected) {
+		t.Errorf("Collect.List returned %+v, expected %+v", collects, expected)
+	}
+}
+
+func TestCollectCount(t *testing.T) {
+	setup()
+	defer teardown()
+
+	httpmock.RegisterResponder("GET", "https://fooshop.myshopify.com/admin/collects/count.json",
+		httpmock.NewStringResponder(200, `{"count": 5}`))
+
+	httpmock.RegisterResponder("GET", "https://fooshop.myshopify.com/admin/collects/count.json?since_id=123",
```

Comment:
should try to use RegisterResponderWithQuery?
https://github.com/jarcoal/httpmock/blob/v1/transport.go